parser.js

```diff
@@ -497,70 +496,98 @@ class DataflashParser {
+                    }
+                }
+
+                if (i % 1000 === 0) {
+                    const perc = 100 * i / len
+                    self.postMessage({ percentage: perc })
+                }
             }
+
+            return parsed
         }
+
         delete this.messages[name]
-        this.messages[name] = parsed
 
-        self.postMessage({ percentage: 100 })
-        console.log(name, this.messageHasInstances(name) ? 'has instances' : 'has no instances')
-        if (parsed.length && this.messageHasInstances(name)) {
-            const instanceField = this.getInstancesFieldName(name)
-            const instances = {}
-            for (const msg of parsed) {
-                try {
-                    instances[msg[instanceField]].push(msg)
-                } catch (e) {
-                    instances[msg[instanceField]] = [msg]
-                }
-            }
-            if (Object.keys(instances).length === 1) {
-                this.fixDataOnce(name)
-                this.simplifyData(name)
-                this.postData({ messageType: name, messageList: this.messages[name] })
-                return parsed
+        const has_instance = this.messageHasInstances(name)
+        console.log(name, has_instance ? 'has instances' : 'has no instances')
+
+        if (has_instance && ("InstancesOffsetArray" in msg_FMT) && (Object.keys(msg_FMT.InstancesOffsetArray).length > 1)) {
+            // Parse instances
+            for (const [index, offsets] of Object.entries(msg_FMT.InstancesOffsetArray)) {
+                const inst_name = name + '[' + index + ']'
+                this.messages[inst_name] = parse(msg_FMT, offsets)
+                this.postData({ messageType: inst_name, messageList: this.messages[inst_name] })
             }
-            for (const [index, messages] of Object.entries(instances)) {
-                const newName = name + '[' + index + ']'
-                this.messages[newName] = messages
-                this.fixDataOnce(newName)
-                this.simplifyData(newName)
-                this.postData({
-                    messageType: newName,
-                    messageList: this.messages[newName]
-                })
+            self.postMessage({ percentage: 100 })
+
+
+            // Old behavior was to return the un-fixed and un-split data here
+            // Putting something in this array makes WebTools work but its dumb
+            // Could be anything....
+            this.messages[name] = { found_log: true }
+            return this.messages[name]
+        }
+
+        // Parse as single msg
+        this.messages[name] = parse(msg_FMT, msg_FMT.OffsetArray)
+
+        // Add mode string to mode message
+        if (msg_FMT.Name === 'MODE') {
+            const mode_len = this.messages[name].Mode.length
+            this.messages[name].asText = new Array(mode_len)
+            for (let i = 0; i < len; i++) {
```

Comment:
```suggestion
            for (let i = 0; i < mode_len; i++) {
```

Reply:
Yep, none of the WebTools load `MODE` ;)

Reply:
fixed